design/dsl/request_test.go

```diff
@@ -0,0 +1,131 @@
+package dsl_test
+
+import (
+	"testing"
+
+	"reflect"
+
+	"goa.design/goa.v2/design"
+	. "goa.design/goa.v2/design/dsl"
+	"goa.design/goa.v2/eval"
+)
+
+func TestRequest(t *testing.T) {
+	var RequestType = Type("Request", func() {
+		Description("Optional description")
+		Attribute("required", design.String)
+		Attribute("name", design.String)
+		Required("required")
+	})
+	cases := map[string]struct {
+		DSL         func()
+		Assert      func(t *testing.T, req design.DataType)
+		ExpectError bool
+	}{
+		"DefinedUserType": {
+			DSL: func() {
+				Request(RequestType, func() {
+					Required("name")
+				})
+			},
+			Assert: func(t *testing.T, req design.DataType) {
+				v, ok := req.(*design.UserTypeExpr)
+				if !ok || v == nil {
+					t.Errorf("expected request to be a design.UserTypeExpr got %v ", reflect.TypeOf(v))
+					return
+				}
+				t.Log(v.AllRequired())
```

Comment:
Same problem as above I believe, need to run the `Execute` on the attribute so the required field is initialized properly.